### PR TITLE
fix(SQLite): handle SQLite errors.

### PR DESCRIPTION
**Pull request by Enuviel**

### CODE REVIEW

File: AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java [hunk 1]
```diff
@@ -93,9 +94,9 @@ static void insertEvent(String event) {
     contentValues.put(COLUMN_DATA, event);
     try {
       database.insert(EVENT_TABLE_NAME, null, contentValues);
+      willSendErrorLog=false;
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java]: space between =

File: AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java [hunk 2]
```diff
@@ -115,14 +116,14 @@ static void insertEvent(String event) {
     try {
       cursor = database.query(EVENT_TABLE_NAME, new String[] {COLUMN_DATA}, null, null, null,
           null, KEY_ROWID + " ASC", "" + count);
+      willSendErrorLog=false;
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java]: same.

File: AndroidSDK/src/com/leanplum/internal/Request.java [hunk 3]
```diff
@@ -87,6 +87,8 @@
 
   private static ApiResponseCallback apiResponse;
 
+  private static List<Map<String,Object>> localErrors=new ArrayList<>();
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: please hit auto format keys :)

File: AndroidSDK/src/com/leanplum/internal/Request.java [hunk 4]
```diff
@@ -387,9 +392,21 @@ protected Void doInBackground(Void... params) {
   }
 
   private void sendRequests() {
-    List<Map<String, Object>> unsentRequests = getUnsentRequests();
-    List<Map<String, Object>> requestsToSend =
-        removeIrrelevantBackgroundStartRequests(unsentRequests);
+    List<Map<String, Object>> unsentRequests = new ArrayList<>();
+    List<Map<String, Object>> requestsToSend;
+    // Check if we have localErrors, if yes then we will send only errors to the server.
+    if (localErrors.size() != 0) {
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: I wonder if we can abstract this a little bit and reuse it here for the error logic and for the regular events?

File: AndroidSDK/src/com/leanplum/internal/Request.java (hunk 3 above)
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: sorry=)

**Pull request merged**